src/core/client/auth/components/SignIn.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { StatelessComponent } from "react";
+import { Field, Form } from "react-final-form";
+import { OnSubmit } from "talk-framework/lib/form";
+import {
+  composeValidators,
+  required,
+  validateEmail,
+  validatePassword,
+} from "talk-framework/lib/validation";
+import {
+  Button,
+  CallOut,
+  Flex,
+  FormField,
+  InputLabel,
+  TextField,
+  Typography,
+  ValidationMessage,
+} from "talk-ui/components";
+import { View } from "../containers/SignInContainer";
+import * as styles from "./SignIn.css";
+
+interface FormProps {
+  email: string;
+  password: string;
+}
+
+export interface SignInForm {
+  onSubmit: OnSubmit<FormProps>;
+  setView: (view: View) => void;
+  error: string | null;
```

Comment:
This state should come from `finalForm`

Reply:
Above. I'm using this for the error returning from the server. Are you referring to the same thing?